src/region/RegionCreator.tsx

```diff
@@ -110,16 +108,26 @@ export default function RegionCreator({ className, onAbort, onStart, onStop }: P
 
         if (shape) {
             onStop(shape);
-        } else {
-            onAbort();
         }
     };
     const updateDraw = (x: number, y: number): void => {
         const [x2, y2] = getPosition(x, y);
+        const { current: x1 } = positionX1Ref;
+        const { current: y1 } = positionY1Ref;
+
+        // If region size is too small, remove region target and reset creator
+        if (!x1 || !y1 || (Math.abs(x2 - x1) < MIN_SIZE && Math.abs(y2 - y1) < MIN_SIZE)) {
```

Comment:
I don't think we should be firing lifecycle events directly in `updateDraw`. This component is complex and is tuned for performance, so we need to be cautious with our changes.

Reply:
Could we detect if the movement is within the threshold and avoid setting `regionDirtyRef.current` to `true`? It seems like that would avoid a render of the rect